hw/ip/edn/doc/_index.md

```diff
@@ -0,0 +1,171 @@
+---
+title: "EDN HWIP Technical Specification"
+---
+
+# Overview
+
+This document specifies EDN hardware IP functionality.
+This module conforms to the [Comportable guideline for peripheral functionality.]({{< relref "doc/rm/comportability_specification" >}})
+
+
+## Features
+
+The Entropy Distribution Network (EDN) block provides both hardware and software interfaces to the [CSRNG IP]({{< relref "hw/ip/csrng/doc" >}}) module.
+Simpler hardware interfaces are provided so that a peripheral block does not have to directly interface with the CSRNG module.
+- A set of registers is provided for firmware to manage a CSRNG application port.
+- Hardware interfaces support four request/acknowledge interfaces, or can be configured to act as a broadcast type of interface (only 1 of 4 interfaces).
+- Hardware interface supports a parameterized data delivery width, for powers of 2 from 1 to 128-bit widths.
+- Auto request mode where generate and reseed commands can be programmed to be done continuously in hardware.
+- Boot mode where a basic instantiate and generate command are done with one TLUL configuration write.
+- Interrupts that are supported:
+  - CSRNG application command has completed.
+  - An internal FIFO error has occurred.
+- No alerts are supported by this revision.
+
+
+## Description
+
+This IP block acts as a gasket between requesting peripheral blocks and the CSRNG block.
+One function this IP block performs is to translate data transfer size.
+For example, CSRNG will return 128 bits on the `genbits` bus.
+A peripheral block may connect with a 32-bit data bus.
+The EDN block will move the first 32 bits from the returned `genbits` bus, and hold on to the remaining data until another request ask for more data. 
+Further, if data is not consumed immediately, the interface to the CSRNG will indicate back pressure to the CSRNG block.
+Each of the four interfaces can request data such that any `genbits` bus return can supply any requesting peripheral block.
+
+A configurable feature for peripheral port 0 is that it can be put into broadcast mode.
+This means that the EDN block will request for that port and is paced by a configurable timer.
+Data will be delivered continuously on the bus for any snooping participant to consume.
```

Comment:
I'm not sure I understand this part. I think you mean that if port 0 is in broadcast mode, it sends data without needing requests. Maybe something like this? "In broadcast mode, the EDN block will generate data on the port at fixed intervals, without needing a request from the consumer. The length of these intervals is configurable."

Reply:
Rewrote this section using your thoughts. It should be clearer now:
A configurable feature for peripheral port 0 is that it can be put into broadcast mode.
When peripheral port 0 is in broadcast mode, the EDN block will make a CSRNG request for that port without requiring a peripheral module request.
Once the `genbits` data is returned to the EDN block from CSRNG, it forwards the response to the peripheral port 0 data bus.
A configurable timer will determine the rate at which `genbits` data will be delivered on the broadcast bus.
Several peripherals are allowed to connect to the broadcast bus.
Each individual peripheral will decide whether to take the data when the ack signal is active.